### PR TITLE
[MXNET-316] Remove empty buckets causing index errors

### PR DESCRIPTION
**Pull request by harusametime**
## Description ##
As reported in the issue #5433, empty buckets cause index errors in mxnet.rnn.BucketSentenceIter.
I modified rnn/io.py to remove the empty buckets as suggested by @hyxie6, then the code works fine.

### CODE REVIEW

**Comment by ThomasDelteil** [comment]
Thanks @harusametime ! Could we get a test to make sure it works and we don't introduce it this bug in the future?

File: python/mxnet/rnn/io.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def __init__(self, sentences, batch_size, buckets=None, invalid_label=-1,
             buff[:len(sent)] = sent
             self.data[buck].append(buff)
 
-        self.data = [np.asarray(i, dtype=dtype) for i in self.data]
+        self.data = [np.asarray(i, dtype=dtype) for i in self.data if i]
```
Comment (ThomasDelteil) [inline python/mxnet/rnn/io.py]: please add a test validating your changes

**Comment by harusametime** [comment]
@ThomasDelteil  Thank you for your comment. I created a ticket and added the label [MXNET-316] to the title of my PR.

**Comment by asitstands** [comment]
https://github.com/apache/incubator-mxnet/blob/master/tests/python/train/test_bucketing.py has tests for BucketSentenceIter. You can add your test there. To execute the test, install nosetests and run `nosetests -v path_to_the_test`.

**Comment by harusametime** [comment]
@asitstands Thanks. That must be helpful for me, but I am not sure what exactly I have to do for testing.  I also need to modify `test_bucketing.py` to check whether my code can work for empty buckets, which have not been considered in `test_bucketing.py`. Then I think what I have to do are
- Modify  and execute `test_bucketing.py` locally
- Report the result of the test here?
- Pull request of new `test_bucketing.py`

Is it correct?

**Comment by asitstands** [comment]
Modify `test_bucketing.py` and test it locally, then push it to your repository (the branch of this PR). It will modity this PR and run all tests, including the one you add, again. To see the test results, click "show all checks - details" below.

**Comment by harusametime** [comment]
@ThomasDelteil Thanks to @asitstands, I could push a test code into my branch. Now I find that all checks have passed. Could you merge my branch into master?

**Comment by ThomasDelteil** [comment]
Thanks @harusametime ! I do not have merge rights. So if I understand correctly you modified the test so that sequences of length=40 are never generated? It might be clearer for the reader if you do `randint(1, max(buckets)-1) # leave out the two last buckets emtpy`

Do you mind commenting on what you updated in the previous test? I see you removed an `assert` statement

**Comment by harusametime** [comment]
@ThomasDelteil I agree to your suggestion on making two buckets empty with `-1`. The previous test evaluates the accuracy by MSE and uses `asserts` for guaranteeing high accuracy. However, the metric should be Perplexity for multiclass classification instead of MSE. Then it turns out the accuracy is not so good because this test requires RNN to forecast random sequence of words. We could not use `asserts` for the accuracy. I will update my test with the comments soon.

**Pull request merged**